--- a/src/main.c	2011-05-03 19:49:15.437957029 +0200
+++ b/src/main.c	2011-05-03 19:49:43.551291297 +0200
@@ -505,7 +505,7 @@
 	maxextents = extents;
 
     decor_quads_to_property(data, GDK_PIXMAP_XID(d->pixmap),
-			    &extents, &maxextents, 0, 0, quads, nQuad);
+			    &extents, &extents, &maxextents, &maxextents, 0, 0, quads, nQuad);
 
     gdk_error_trap_push();
     XChangeProperty(xdisplay, d->prop_xid,
@@ -1937,7 +1937,7 @@
     nQuad = set_switcher_quads(quads, d->width, d->height, ws);
 
     decor_quads_to_property(data, GDK_PIXMAP_XID(d->pixmap),
-			    &extents, &extents, 0, 0, quads, nQuad);
+			    &extents, &extents, &extents, &extents, 0, 0, quads, nQuad);
 
     style = gtk_widget_get_style (style_window);
 
@@ -2364,7 +2364,7 @@
 	nQuad = set_shadow_quads(quads, width, height, ws);
 
 	decor_quads_to_property(data, GDK_PIXMAP_XID(ws->shadow_pixmap),
-				&ws->shadow_extents, &ws->shadow_extents, 0, 0,
+				&ws->shadow_extents, &ws->shadow_extents, &ws->shadow_extents, &ws->shadow_extents, 0, 0,
 				quads, nQuad);
 
 	XChangeProperty(xdisplay, xroot,
@@ -2424,7 +2424,7 @@
 	(*d.draw) (&d);
 
 	decor_quads_to_property(data, GDK_PIXMAP_XID(d.p_inactive),
-				&extents, &extents, 0, 0, quads, nQuad);
+				&extents, &extents, &extents, &extents, 0, 0, quads, nQuad);
 
 	XChangeProperty(xdisplay, xroot,
 			normalAtom,
@@ -2433,7 +2433,7 @@
 			BASE_PROP_SIZE + QUAD_PROP_SIZE * nQuad);
 
 	decor_quads_to_property(data, GDK_PIXMAP_XID(d.p_active),
-				&extents, &extents, 0, 0, quads, nQuad);
+				&extents, &extents, &extents, &extents, 0, 0, quads, nQuad);
 
 	XChangeProperty(xdisplay, xroot,
 			activeAtom,
@@ -5512,7 +5512,7 @@
     gdkdisplay = gdk_display_get_default();
     xdisplay = gdk_x11_display_get_xdisplay(gdkdisplay);
 
-    frame_window_atom = XInternAtom(xdisplay, "_NET_FRAME_WINDOW", FALSE);
+    frame_window_atom = XInternAtom(xdisplay, DECOR_INPUT_FRAME_ATOM_NAME, FALSE);
     win_decor_atom = XInternAtom(xdisplay, DECOR_WINDOW_ATOM_NAME, FALSE);
     win_blur_decor_atom = XInternAtom (xdisplay, DECOR_BLUR_ATOM_NAME, FALSE);
     wm_move_resize_atom = XInternAtom(xdisplay, "_NET_WM_MOVERESIZE", FALSE);
@@ -5617,7 +5617,8 @@
     update_window_extents(ws);
     update_shadow(pfs);
 
-    decor_set_dm_check_hint(xdisplay, DefaultScreen(xdisplay));
+    decor_set_dm_check_hint(xdisplay, DefaultScreen(xdisplay),
+    			    WINDOW_DECORATION_TYPE_PIXMAP);
 
     update_settings(ws);
 
